### PR TITLE
DESCW-772 integrated custom theme options

### PR DESCRIPTION
**Pull request by codewisenate**
- Refactored setup with default values to make use of new options (uses WordPress Settings API)
- Built custom theme options page to update settings, tied into Appearance menu along side theme documentation 
- Minor display fixes for menu and card styles (impacts Symposium site)
- Note: after 1.0.3 default options will need to be shifted to BCGov standard, currently set to CleanBC settings so when moved to production that site will not need immediate triage.

### CODE REVIEW

File: inc/core/templates/theme-options-page.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+*
+* @since 1.0.3
+*/
+?>
+
+
+<div class="bcgov-block-theme-page-header">
+	<div class="bcgov-block-theme-page-header__container">
+		<div class="bcgov-block-theme-page-header__branding">
+		<h2>BCGov Block Theme Site Options</h2>
```
Comment (Baelx) [inline inc/core/templates/theme-options-page.php:33]:
```suggestion
		<h2><?php esc_html_e('BCGov Block Theme Site Options'); ?></h2>
```

File: inc/core/templates/theme-options-page.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+/**
+ * Theme admin functions.
+ *
+ * @package BCGov Block Theme
+ */
+
+namespace Bcgov\Theme\Block;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit( '<h1>Direct access denied.</h1>' );
```
Comment (Baelx) [inline inc/core/templates/theme-options-page.php:11]: Don't know if it's necessary for us to present a multi-lingual experience to those attempting direct access but:

```suggestion
        $exit_msg = esc_html_e('Direct access denied.');
	exit( '<h1>' . $exit_msg . '</h1>' );
```

File: inc/core/templates/theme-options-page.php [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+					
+					<p class="bcgov-block-theme-intro-text">
+						<?php echo esc_html__( 'The following settings will allow adjusting specific behaviours of BCGov Block Theme.', 'bcgov-block-theme' ); ?>
+					</p>
+				</div> <!-- /Getting started -->
+
+				<div class="bcgov-block-theme-grid-content card">
+					<h3>Active Site Pattern Styles</h3>
+					
+					<p class="description">Choose the active site – this choice will impact how certain patterns display and which patterns and blocks may or may not be available.</p>
```
Comment (Baelx) [inline inc/core/templates/theme-options-page.php:66]:
```suggestion
					<p class="description"><?php esc_html_e('Choose the active site – this choice will impact how certain patterns display and which patterns and blocks may or may not be available.'); ?></p>
```

File: src/Setup.php [hunk 4]
```diff
@@ -299,13 +260,23 @@ public function bcgov_block_theme_dependencies() {
 	 * @return void
 	 */
 	public function bcgov_block_theme_menu() {
+
+		add_theme_page(
+			esc_html__( 'Using BCGov Block Theme', 'bcgov-block-theme' ),
+			esc_html__( 'Using the BCGov Block Theme', 'bcgov-block-theme' ),
+			'edit_theme_options',
+			'bcgov-block-theme-docs',
+			[ &$this, 'bcgov_block_theme_documentation_display' ]
```
Comment (Baelx) [inline src/Setup.php:269]: Is this supposed to be `$this` ?

File: src/Setup.php [hunk 5]
```diff
@@ -299,13 +260,23 @@ public function bcgov_block_theme_dependencies() {
...
 		add_theme_page(
-			esc_html__( 'BCGov Blocks Theme', 'bcgov-block-theme' ),
-			esc_html__( 'Using BCGov Blocks Theme', 'bcgov-block-theme' ),
+			esc_html__( 'BCGov Block Theme Options', 'bcgov-block-theme' ),
+			esc_html__( 'Theme Options', 'bcgov-block-theme' ),
 			'edit_theme_options',
-			'bcgov-block-theme-info',
-			[ &$this, 'bcgov_block_theme_page_display' ]
+			'bcgov-block-theme-options',
+			[ &$this, 'bcgov_block_theme_options_display' ]
```
Comment (Baelx) [inline src/Setup.php:277]: Here too

File: src/styles/public/navigation.scss [hunk 6]
```diff
@@ -246,6 +246,7 @@ body.has-ribbon-menu {
 		&__container {
 			flex-direction: row !important;
```
Comment (Baelx) [inline src/styles/public/navigation.scss:247]: Just curious about the use of `!important` here. Are you fighting against some other inaccessible styles?

File: src/styles/public/navigation.scss [hunk 7]
```diff
@@ -280,16 +282,19 @@ body.has-ribbon-menu {
 				// border: 1px dashed lime;
```
Comment (Baelx) [inline src/styles/public/navigation.scss:282]: Will you need this commented out CSS later?

**Review by Baelx — APPROVED** [review]
Looks good! Not sure about how necessary translatable strings are for this project but I noticed a few instances of static text are missing i18n. 

Also, maybe have a look at `&$this` - I've never seen this before. Is it a typo?

I'm curious about the use of `!important` in the CSS. Is this a consequence of block patterns and fighting against native block styling?

**Review by ShawnTurple — APPROVED** [review]

File: inc/core/templates/theme-options-page.php (hunk 2 above)
Comment (codewisenate) [reply inc/core/templates/theme-options-page.php:11]: Added: used the return version rather than echo.

Comment (codewisenate) [reply inc/core/templates/theme-options-page.php:11]: Valid point. Removed translation functions on these pages. Decreases maintenance complexity.

**Pull request merged**